### ISSUE TITLE
pattern for defining Enumerations

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/normal-modes#34]
Related to #2 (code review):

> - [ ] Type expressions for Enumeration values should be annotated as instances of that Enumeration, see examples in https://github.com/phetsims/phet-core/blob/master/js/Enumeration.js for more. ...

There's a problem with both the type expression for `EnumerationProperty` and the definitions of `Enumerations` in general.

Example in TwoDimensionsModel.js:

```js
      // @public {Enumeration}
      this.ampSelectorAxis = Enumeration.byKeys( [ 'HORIZONTAL', 'VERTICAL' ] );

      // @public {Property.<string>} the current direction of motion of the visible masses
      this.ampSelectorAxisProperty = new EnumerationProperty( this.ampSelectorAxis, this.ampSelectorAxis.VERTICAL, {
        tandem: tandem.createTandem( 'ampSelectorAxisProperty' )
      } );
```

The lefthand side of an Enumeration definition should look like a class name.  And we typically put them in their own .js file.  For example, the above would be:

```
define( require => {
  'use strict';

  // modules
  const Enumeration = require( 'PHET_CORE/Enumeration' );
  const normalModes = require( 'NORMAL_MODES/normalModes' );

  const AmplitudeSelectorAxes = Enumeration.byKeys( [ 'HORIZONTAL', 'VERTICAL' ] );

  return phScale.register( 'AmplitudeSelectorAxes', AmplitudeSelectorAxes );
} );
```

Then the definition of `this.ampSelectorAxisProperty` would look like this.  Note the type expression, which can be optionally omitted because it's obvious from `new EnumerationProperty( AmplitudeSelectorAxes`. 

```js
      // @public {EnmerationProperty.< AmplitudeSelectorAxes >} the current direction of motion of the visible masses
      this.ampSelectorAxisProperty = new EnumerationProperty( AmplitudeSelectorAxes, AmplitudeSelectorAxes.VERTICAL, {
        tandem: tandem.createTandem( 'ampSelectorAxisProperty' )
      } );
```

This same pattern appears in OneDimensionModel.js.

Recommendation:
* Factor out Enumerations into their own .js files.
* Adjust code that uses these Enumerations

**Comment by pixelzoom** [comment]
With assertions enabled `?ea`, this is also causing the following failure:

```
assert.js?bust=1581370390829:22 Uncaught Error: Assertion failed: value is not a member of Enumeration [object Object]
    at window.assertions.assertFunction (assert.js?bust=1581370390829:22)
    at Object.isValueValidValueType (ValidatorDef.js?bust=1581370390911:332)
    at Object.isValueValid (ValidatorDef.js?bust=1581370390911:260)
    at Object.isValueValid (ValidatorDef.js?bust=1581370390911:307)
    at Object.isValidValidator (ValidatorDef.js?bust=1581370390911:144)
    at Object.validateValidator (ValidatorDef.js?bust=1581370390911:198)
    at new Property (Property.js?bust=1581370390911:139)
    at new EnumerationProperty (EnumerationProperty.js?bust=1581370390911:48)
    at new TwoDimensionsModel (TwoDimensionsModel.js?bust=1581370390911:152)
    at TwoDimensionsScreen.createModel (TwoDimensionsScreen.js?bust=1581370390911:35)
```

**Comment by pixelzoom** [comment]
The above error seems to be a PhET-iO problem.  It looks like there's an assumption that keys are unique across all Enumerations.     I factored out a single `DirectionOfMotion` enumeration, and the error went away.

**Comment by zepumph** [comment]
Over in https://github.com/phetsims/phet-core/issues/79#issuecomment-584429471 I was able to commit a fix for this error as it pertains to using phetioType as validation for EnumerationProperties. 

I have two thoughts:
1. https://github.com/phetsims/normal-modes/commit/6b48cf512d9a4ccc48a92c8251d8cf633ad3f08e feels like the right direction to me. I think it is rare to have two Enumerations that have the same keys, and factoring that out to its own file feels good to me.
2. I fixed this in the other issue anyways, and tested by locally reverting https://github.com/phetsims/normal-modes/commit/6b48cf512d9a4ccc48a92c8251d8cf633ad3f08e and then running the sim with no problems. Then when I commented out my fix, I got the error again. FYI it is enough to just comment out this line in EnumerationIO: 

 `    EnumerationIOImpl.cacheKey = enumeration;`


@pixelzoom let me know if I can be of more assistance.
